src/main/java/io/github/christophermanahan/carnitas/LoggingMiddleware.java

```diff
@@ -0,0 +1,18 @@
+package io.github.christophermanahan.carnitas;
+
+public class LoggingMiddleware implements Handler {
+    private final MessageLogger logger;
+    private final Handler handler;
+
+    LoggingMiddleware(Handler handler) {
+        this.logger = new MessageLogger();
+        this.handler = handler;
+    }
+
+    public Response handle(Request request) {
+        logger.log(request.toString());
```

Comment:
Usually `toString` is used for debugging purposes, not for presentation logic.

Also, we don't need the whole request in the log. For the request, it is fine to just have something like

```
METHOD /path HTTP/1.1
```
and for response

```
200 OK
```